test/framework/k8s_utils.go

```diff
@@ -371,6 +373,24 @@ func DownloadKubeconfig(ctx context.Context, client kubernetes.Interface, namesp
 	return nil
 }
 
+// DownloadAdminKubeconfigForShoot requests an admin kubeconfig for the given shoot and writes the kubeconfig to the
+// given download path. The kubeconfig expires in 24 hours.
+func DownloadAdminKubeconfigForShoot(ctx context.Context, client kubernetes.Interface, shoot *gardencorev1beta1.Shoot, downloadPath string) error {
+	const expirationSeconds = 24 * 3600 // 24h
```

Comment:
Currently we know about the following tests using this `$TM_KUBECONFIG_PATH/shoot.config`:
- conformance tests - https://github.com/gardener/test-infra/blob/38688055e0812a5b5fb802827fb761c4e55ba146/.test-defs/conformanceTestgrid.yaml#L56 - it has `activeDeadlineSeconds: 10800` (3h)
- MCM integration tests - https://github.com/gardener/machine-controller-manager-provider-aws/blob/2493c58a03b991fb1faf865c3cf1082436410b2b/.test-defs/IntegrationTest.yaml#L13 - it has `activeDeadlineSeconds: 1200` (20min)
- shoot-dns intergration tests - https://github.com/gardener/gardener-extension-shoot-dns-service/blob/d6ff1258333000449904ab32ea41bcc5d09a4804/.test-defs/shootdns-test.yaml#L17 - it has   `activeDeadlineSeconds: 1800` (30min)
- backup-restore integration tests - https://github.com/gardener/etcd-backup-restore/blob/9a8bdad14a86619e1791544714722a0e9eeccaf1/.ci/performance_regression_test#L41 - it has `activeDeadlineSeconds: 1200` (20min)
- remedy-controller integration tests - https://github.com/gardener/remedy-controller/blob/703570317b235d24717ece5c39d77daf9b67ed4d/.ci/integration_test.py#L45 - it has `activeDeadlineSeconds: 2700` (45 min)

I suggest to use 4h. Is that okay for everyone? 

cc @hendrikKahl

Reply:
Sounds good. Do we need to document this requirement for the Gardener environment the tests run against somewhere?

Reply:
Since you can run different test suits in sequentially in testmachinery, I would prefer an expiration time of something like 5-6h to be on the safe side.

Reply:
> Sounds good. Do we need to document this requirement for the Gardener environment the tests run against somewhere?

If my understanding of the whole flow is correct, then it should be useful to write down a sentence it as done in https://github.com/gardener/test-infra/pull/441.

> Since you can run different test suits in sequentially in testmachinery, I would prefer an expiration time of something like 5-6h to be on the safe side.

Nice catch! You are right. I now set it to 6h. I am also open to extend it even further (to 8h for example) if you see the need for this.

Reply:
Do we want to go with 6h?

Reply:
I interpret the silence as everyone is fine with 6h.